src/components/DataTable.vue

```diff
@@ -6,69 +6,99 @@
           <th>Sr#</th>
           <th>Author</th>
           <th>Title</th>
+          <th>Actions</th>
         </tr>
       </thead>
       <tbody>
-        <tr v-bind:key="tableItem + index" v-for="(tableItem, index) in tableData">
+        <tr
+          v-bind:key="postItem + index"
+          v-for="(postItem, index) in postsData"
+        >
           <td>{{ index + 1 }}</td>
-          <td>{{ tableItem.data.author }}</td>
-          <td>{{ tableItem.data.title }}</td>
+          <td>{{ postItem.data.author }}</td>
+          <td>{{ postItem.data.title }}</td>
+          <td>
+            <button
+              class="btn btn-primary"
+              @click="showRowDetail(postItem.data)"
+            >
+              Detail
+            </button>
+          </td>
         </tr>
       </tbody>
     </table>
+    <PostList
+      v-on:deletePost="deletePost($event)"
+      :post-details="postDetails"
+    />
   </div>
 </template>
 
 <script>
 import axios from "axios";
 import $ from "jquery";
+import PostList from "./PostList.vue";
 
 export default {
   name: "Data-Table",
+  components: {
+    PostList,
+  },
+
   data() {
     return {
-      tableData: []
+      postsData: [],
+      showDetail: false,
+      postDetails: [],
     };
   },
 
-  methods:{
-    
-    searchDelay:()=>{
+  methods: {
+    showRowDetail(tableItem) {
+      this.postDetails.push(tableItem);
+    },
 
-    let myDataTable;
-    $(document).ready(function() {
-      myDataTable = $(".data-table")
-        .dataTable()
-        .api({
-          data: this.tableData,
-          columns: [{ title: "sr#" }, { title: "Author" }, { title: "Title" }],
-        
-      
-        });
+    deletePost(data) {
+      this.postsData = this.postsData.filter((currentItem) => {
```

Comment:
We can avoid brackets `()` here in callback argument. 
Also we can avoid `return` keyword and `{}`.If we have only one line of action in implementation.